File: Data Structures & Algorithms/combination-target-sum/submission-2.cpp
class Solution {
public:
    void solve(vector<int>& nums,int target,int ind,vector<vector<int>>&res,vector<int>&temp){
        if (target == 0){
            res.push_back(temp);
        }
        if (ind == nums.size()){
            return;
        }
        if (target < nums[ind]){
            return;
        }
        temp.push_back(nums[ind]);
        solve(nums,target-nums[ind],ind,res,temp);
        temp.pop_back();
        solve(nums,target,ind+1,res,temp);

    }
    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
        sort(nums.begin(),nums.end());  
        vector<vector<int>>res;
        vector<int>temp;
        solve(nums,target,0,res,temp);
        return res; 
    }
};

File: Data Structures & Algorithms/house-robber/submission-2.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        dp = [0 for i in range(len(nums)+1)]
        if (len(nums)==1): return nums[0]
        if (len(nums)==2):return max(nums[0],nums[1])
        dp[-2]= nums[-1]
        for i in range(len(nums)-2,-1,-1):
            dp[i] = max(dp[i+1],nums[i] + dp[i+2])
        return max(dp[0],dp[1])